decoder/stats.go

```diff
@@ -365,6 +412,156 @@ func updatePokemonStats(old *Pokemon, new *Pokemon, areas []geo.AreaName) {
 	}
 }
 
+func updateRaidStats(old *Gym, new *Gym, areas []geo.AreaName) {
+	if len(areas) == 0 {
+		areas = []geo.AreaName{
+			{
+				Parent: "world",
+				Name:   "world",
+			},
+		}
+	}
+
+	locked := false
+
+	// Loop though all areas
+	for i := 0; i < len(areas); i++ {
+		area := areas[i]
+
+		// Check if RaidSpawnTimestamp has changed, then we can assume raid has disappeared or new raid started.
+		if old == nil || old.RaidSpawnTimestamp != new.RaidSpawnTimestamp {
```

Comment:
If a new gym is found, then it will run until the new.RaidPokemonId.Int64 != 0 check yeah, and add the matching area to raidCount.

Have tried with truncated gym table multiple times to trigger the flow faster. Havn't seen any panic caused by that, so it should be bare minimal in memory, until it gets a gym with a RaidPokemonId